### PR TITLE
build(aio): upgrade to codelyzer(4.0.x)

### PR DESCRIPTION
**Pull request by akkumar**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

Upgrades the boilerplate sample in cli to be codelyzer 4.0.0 instead of 3.2.x. 

## What is the current behavior?

Currently it is at 
```json
    "codelyzer": "~3.2.0",
```
Issue Number: angular/angular-cli#8463

## What is the new behavior?

codelyzer for ng5 should refer to ~4.0.1
Currently it is at 
```json
    "codelyzer": "^4.0.1",
```

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

More details at - https://github.com/mgechev/codelyzer/blob/master/CHANGELOG.md#400 .

### CODE REVIEW

File: aio/tools/examples/shared/boilerplate/cli/package.json [hunk 1]
```diff
@@ -32,7 +32,7 @@
     "@types/jasmine": "~2.5.53",
     "@types/jasminewd2": "~2.0.2",
     "@types/node": "~6.0.60",
-    "codelyzer": "~3.2.0",
+    "codelyzer": "^4.0.0",
```
Comment (gkalpak) [inline aio/tools/examples/shared/boilerplate/cli/package.json]: This is supposed to mirror what the `@angular/cli` would generate and right now, it [would generate][1] `~3.2.0`, so I think it is best to leave it at that.

[1]: https://github.com/angular/devkit/blob/079f472781f9ce9c48ee5a88d781a35597bc1d6c/packages/schematics/angular/application/files/package.json#L36

Comment (akkumar) [reply aio/tools/examples/shared/boilerplate/cli/package.json]: @gkalpak - Thanks for pointing to the right source of truth. angular/devkit#272 logged independently to address the same. 

We can visit this later after the source of truth in devkit is addressed. Thanks !

Comment (akkumar) [reply aio/tools/examples/shared/boilerplate/cli/package.json]: angular/devkit#253 and  angular/devkit#257 under discussion for the same issue as well. FYI

Comment (akkumar) [reply aio/tools/examples/shared/boilerplate/cli/package.json]: angular/devkit#257  commit - https://github.com/angular/devkit/commit/3a620e2bd3d9549fce7e8c479e5f9b866dcd7a3a is now in, making codelyzer at 4.0.1 

So - now may this issue can be merged as well , @gkalpak ?

Comment (gkalpak) [reply aio/tools/examples/shared/boilerplate/cli/package.json]: `@angular/cli` is still at 1.5.0 in this package.json (and the merged PR hasn't been released yet afaict), so we con't merge it. To fix it in the examples, we need to upgrade the `@angular/cli` dependency and the rest of `package.json` to match what would have been generated with the cli.

Comment (akkumar) [reply aio/tools/examples/shared/boilerplate/cli/package.json]: @angular/cli 1.5.4 contains the fix for commit under discussion ( to ugprade codelyzer to 4.0.1 by default). 

```console
$ cat package.json | grep codelyzer
    "codelyzer": "^4.0.1",
```
```console
 $ ng version
Angular CLI: 1.5.4
```


upgrading angular/cli to 1.5.4 as well in this package.json to reflect the change, hence.

**Comment by akkumar** [comment]
Did a force commit and lost the previous conversation history here (associated with one of the earlier commits). 

summary: angular/cli 1.5.4 released recently contains codelyzer ^4.0.1 by default as intended. So upgrading angular/cli and codelyzer as appropriate in the package.json . 

cc @gkalpak

**Review by gkalpak — APPROVED** [review]

**Pull request closed without merging**